### PR TITLE
Gcp secret manager migrate to new module

### LINKED ISSUE #515: Migrate `GCPSecretManagerConfigStore` to `@configu/integrations`
### Suggestion

Migrate the `GCPSecretManagerConfigStore` from its current setup to the new `@configu/integrations` module as part of the major restructuring. The goal is to move all third-party integrations, including stores, expressions, and libraries, to `@configu/integrations`.

### Motivation

This migration is part of a larger effort to better organize and modularize the product. Moving the `GCPSecretManagerConfigStore` will align with the new structure, making the system cleaner and more maintainable.

### Context

You can refer to the already migrated stores for guidance:  
[Migrated stores in integrations](https://github.com/configu/configu/tree/main/packages/integrations/stores).

Base stores implementations from `@configu/ts` can be found here:  
[Base stores in TS](https://github.com/configu/configu/tree/52cee9c41fb03addc4c0983028e37df42945f5b7/packages/ts/src/stores).

The old node-based stores were previously in `@configu/node`:  
[Node stores](https://github.com/configu/configu/tree/52cee9c41fb03addc4c0983028e37df42945f5b7/packages/node/src/stores).

The old browser-based stores were previously in `@configu/browser`:  
[Browser stores](https://github.com/configu/configu/tree/52cee9c41fb03addc4c0983028e37df42945f5b7/packages/browser/src/stores).

### PR DESCRIPTION
**Pull request by Itamarled**
Closes #515

### CODE REVIEW

File: packages/integrations/stores/gcp-secret-manager-config-store/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "homepage": "https://configu.com",
+  "keywords": [
+    "cfgu",
+    "configu"
+  ],
+  "scripts": {
+    "build": "tsup"
+  },
+  "dependencies": {
+    "@aws-sdk/client-ssm": "^3.667.0",
```
Comment (davidmeirlevy) [inline packages/integrations/stores/gcp-secret-manager-config-store/package.json:25]: i don't think you need the aws-sdk on gcp integration.

File: packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager-config-store.ts
Comment (davidmeirlevy) [inline packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager-config-store.ts:1]: 1. where is the content of the store?
2. don't use "config-store" suffix on the file name. just the name of the store itself ("gcp-secret-manager.ts").

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]
Thanks for your PR. 👍 

it still needs some work to be done before we can merge it.

File: packages/integrations/stores/gcp-secret-manager-config-store/package.json (hunk 1 above)
Comment (Itamarled) [reply packages/integrations/stores/gcp-secret-manager-config-store/package.json:25]: I think you are right.
should there be a dependency related to gcp to add?

Comment (davidmeirlevy) [reply packages/integrations/stores/gcp-secret-manager-config-store/package.json]: yes, this one:
```import { SecretManagerServiceClient } from '@google-cloud/secret-manager';```

File: packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import { SecretManagerServiceClient } from '@google-cloud/secret-manager';
```
Comment (davidmeirlevy) [inline packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts:1]: you should add this package as a dependency.

File: packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import { KeyValueConfigStore } from '@configu/ts';
```
Comment (davidmeirlevy) [inline packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts:2]: `@configu/ts` has been replaced by `@configu/sdk`.

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]
two more comments. 👍

File: packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts (hunk 2 above)
Comment (Itamarled) [reply packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts:1]: @davidmeirlevy 
Found the version in the package.json from the old store json file.
seems like according to the checks it's not the same as the one specified in the validation.
where can I see the correct version for this dependency?

Comment (davidmeirlevy) [reply packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts:1]: "@google-cloud/secret-manager": "^5.4.0",

File: packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts (hunk 3 above)
Comment (davidmeirlevy) [reply packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts]: sorry  for missleading, thenew version of KeyValueConfigStore should come from:
```import { KeyValueConfigStore } from '@configu/integrations/src/utils/KeyValue';```

File: packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts (hunk 2 above)
Comment (Itamarled) [reply packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts:1]: so the other import was the issue maybe

File: packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts (hunk 3 above)
Comment (Itamarled) [reply packages/integrations/stores/gcp-secret-manager-config-store/src/gcp-secret-manager.ts]: fixed

**Review by davidmeirlevy — APPROVED** [review]

**Comment by davidmeirlevy** [comment]
@Itamarled approved!
Thanks!

**Comment by rannn505** [comment]
@Itamarled Thanks man, thats awesome work! 💘

**Comment by pelegpor** [comment]
Hi @Itamarled ,

Here are the details for claiming your Hacktoberfest prizes! 🎉 Check out [#465  (comment)](https://github.com/configu/configu/issues/465#issuecomment-2498550259) for more information or simply fill out the form here: [Claim Your Swag](https://share-eu1.hsforms.com/2SVc5VKY3Thi04Bvjh17WKw2e62o0).  

Remember, you only need to fill out the form once for all your PRs. 😊

**Pull request merged**